lib/datadog/open_feature/exposures/worker.rb

```diff
@@ -0,0 +1,125 @@
+# frozen_string_literal: true
+
+require_relative '../../core/utils/time'
+require_relative '../../core/workers/queue'
+require_relative '../../core/workers/polling'
+
+require_relative 'buffer'
+require_relative 'batch_builder'
+
+module Datadog
+  module OpenFeature
+    module Exposures
+      # This class is responsible for sending exposures to the Agent
+      class Worker
+        include Core::Workers::Queue
+        include Core::Workers::Polling
+
+        GRACEFUL_SHOTDOWN_EXTRA_SECONDS = 5
+        GRACEFUL_SHOTDOWN_WAIT_INTERVAL_SECONDS = 0.5
+
+        DEFAULT_FLUSH_INTERVAL_SECONDS = 30
+        DEFAULT_BUFFER_LIMIT = Buffer::DEFAULT_LIMIT
+
+        attr_reader :logger
+
+        def initialize(
+          settings:,
+          transport:,
+          logger: Datadog.logger,
```

Comment:
(Similar note as in other spots)

```suggestion
          logger:,
```